core/finality-grandpa/src/until_imported.rs

```diff
@@ -64,7 +65,7 @@ pub(crate) trait BlockUntilImported<Block: BlockT>: Sized {
 
 /// Buffering imported messages until blocks with given hashes are imported.
 pub(crate) struct UntilImported<Block: BlockT, Status, I, M: BlockUntilImported<Block>> {
-	import_notifications: Fuse<ImportNotifications<Block>>,
+	import_notifications: Fuse<Box<dyn Stream<Item = BlockImportNotification<Block>, Error = ()> + Send>>,
```

Comment:
Can you not just put the `Fuse` into the `Box` and prevent the huge number of type annotations below?

Reply:
No, because we're calling `is_done()` somewhere else.